### ISSUE TITLE
Intrinsics on Mixed-Images

### ISSUE THREAD

**Issue by AlexandreDiPiazza** [yanglei18/Mix-Teaching#8]
Hi,

Congratulations for the great work. 

I understand that for MonoFlex, each object in the labels has its own intrinsics (matrix Ps in the labels) when computing the Loss.  In the Mixed images, I analysed the matrices Ps of the different objects ( pasted and non pasted) and surprisingly it seems like each object has the same intrinsics ( corresponding matrix Ps).

My question is the following: 
- Some other models, such as DD3D, only take one set of intrinsic parameters in the labels for the entire image. Do you think the mixed dataset could be used in a setting where there is one intrinsics matrix for the whole image, instead of per-object intrinsics matrices in the labels as in MonoFlex?

Thank you.

**Comment by yanglei18** [comment]
Thank you for your attention, In the case of DD3D, which uses a single set of intrinsic parameters for the entire image labels, it is crucial to ensure that the original images used to construct both the background database and instance database are captured using the same camera.

**Comment by AlexandreDiPiazza** [comment]
Thank you for your answer.

Do you think the Mixed-dataset from your code, created with KITTI-3D, satisfies this condition? From my analysis it seemed to satisfy this condition. In that case it should be possible to directly use for example DD3D on the mixed-dataset created by your implementation on KITTI-3D.

**Comment by yanglei18** [comment]
Yes, exactly. In our Mix-Teaching data pipeline, the instances and backgrounds used for image combination come from images with the same resolution, ensuring that they correspond to the same intrinsic parameters.

**Comment by AlexandreDiPiazza** [comment]
Thank you so much for your answers.
I'm eager to try your techniques with DD3D. Have you ever used Mix-Teaching techniques with this model?
At first, I was hesitant because I thought that some of the mixed images
(e.g., 1-  training images with raw instances pasted or 2-  background images
with raw images pasted) might need methods that require intrinsics per
object, instead of one intrinsics per image. But I'm glad to hear that it
should work.

**Comment by yanglei18** [comment]
We have not used  Mix-Teaching for DD3D. However, Mix-Teaching is compatible with the DD3D  on the  KITTI dataset. This compatibility is due to the open-source code utilizing instances and background images from the same camera for the copypaste process.

**Comment by AlexandreDiPiazza** [comment]
Thank you for your answers.
I am also wondering, do you use some test-time augmentations when evaluating results on the validation KITTI-3D dataset during training with : `python tools/plain_train_net.py --batch_size 8 --backbone dla34 --config runs/monoflex.yaml --mix_teaching True --ckpt <path-to-previous-teacher-model> --output output/exp`
